client/app/shared/modals/data-upload/data-upload.component.spec.ts

```diff
@@ -0,0 +1,32 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing'
+
+import { DataUpload } from './data-upload.component'
+import { LoggerService } from '../../services/logger.service'
+import { ErrorHandler } from '@angular/core'
+
+describe('DataUpload', () => {
+  let component: DataUpload
+  let fixture: ComponentFixture<DataUpload>
+  let logger: LoggerService
+  let handler: ErrorHandler
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [DataUpload]
+    })
+    .compileComponents()
+    .catch(err => handler.handleError(err))
+    .then(() => logger.error('Could not compile components'))
+    .catch(() => 'Obligatory catch')
```

Comment:
return the promise instead

Reply:
I don't have a lot of experience with this, how would you change the code?

Reply:
No problem, we are doing those CR to learn. The way to do that is:
```
return TestBed.configureTestingModule({
      declarations: [DataUpload]
    })
    .compileComponents()
    .catch(err => handler.handleError(err))
    .then(() => logger.error('Could not compile components'))
```
This the promise you create after will be returned by the function. This will also cause the test runner to wait for the promise to be fulfilled before continuing. For more information about promises check out https://scotch.io/tutorials/javascript-promises-for-dummies

Reply:
![image](https://user-images.githubusercontent.com/28773375/60399968-1575d200-9b76-11e9-952a-c4c0d8d479ad.png)
By changing it to that it now has a lint error